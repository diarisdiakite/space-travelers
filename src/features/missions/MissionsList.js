import React from 'react';
import { useSelector } from 'react-redux';
import {
  selectAllMissions,
  // joinMission, cancelMissionParticipation,
} from './missionsSlice';
import MissionsListElements from './childComponents/MissionsListElements';

function MissionsList() {
  const missions = useSelector(selectAllMissions);
  console.log(missions);

  return (
    <div className="main-container">
      <MissionsListElements
        missions={missions}
      />
    </div>
  );
}

export default MissionsList;
