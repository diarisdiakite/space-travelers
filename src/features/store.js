import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
