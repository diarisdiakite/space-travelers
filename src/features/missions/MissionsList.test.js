import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import MissionsList from './MissionsList';

describe('MissionsList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <MissionsList />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
