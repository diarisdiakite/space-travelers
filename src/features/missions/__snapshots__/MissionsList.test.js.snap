// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsList component renders correctly 1`] = `
<div
  className="main-container"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
